test/common/http/conn_manager_impl_test.cc

```diff
@@ -3206,6 +3245,97 @@ TEST_F(HttpConnectionManagerImplTest, MultipleFilters) {
   EXPECT_EQ(ssl_connection_.get(), encoder_filters_[1]->callbacks_->connection()->ssl());
 }
 
+// MultipleFiltersWithReversedEncoders verifies the same case of MultipleFilters, except that the
```

Comment:
Are these the only tests that depend on order? If there are others, I think we should convert them to the reverse order and only keep around one or two tests for the existing order.

Reply:
As per off-line discussion, set the default value to true in the test and left two tests (with suffix of .*InorderEncoders) verifying the existing order.